src/containers/my-resources/questions-container/QuestionsContainer.tsx

```diff
@@ -52,106 +46,105 @@ const QuestionsContainer = () => {
+      return response
+    },
+    [queryClient]
   )
 
   const editQuestion = (id: string) => {
-    navigate(createUrlPath(authRoutes.myResources.editQuestion.path, id))
+    return navigate(
+      getFullUrl({
+        pathname: authRoutes.myResources.editQuestion.route,
+        parameters: { id }
+      })
+    )
   }
 
   const duplicateQuestion = useCallback(
-    (id?: string) => {
-      const item = response.items.find(
-        (element) => element._id === id
-      ) as Question
+    async (id: string) => {
+      if (!questions) {
+        handleErrorAlert(DuplicateQuestionErrors.QUESTIONS_NOT_FOUND)
+        return
+      }
 
-      return ResourceService.createQuestion({
+      const item = questions.items.find((element) => element._id === id)
+
+      if (!item) {
+        handleErrorAlert(DuplicateQuestionErrors.QUESTION_NOT_FOUND)
+        return
+      }
+
+      return await ResourceService.createQuestionQuery({
         title: item.title,
         text: item.text,
         answers: item.answers,
         category: item.category?._id ?? null,
         type: item.type
       })
     },
-    [response.items]
+    [questions, handleErrorAlert]
   )
 
-  const onDuplicateError = (error?: ErrorResponse) => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.error,
-        message: getErrorKey(error)
-      })
-    )
-  }
-
   const onDuplicateResponse = () => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.success,
-        message: `myResourcesPage.questions.successDuplication`
-      })
-    )
+    handleAlert({
+      severity: snackbarVariants.success,
+      message: `myResourcesPage.questions.successDuplication`
+    })
   }
 
-  const { error: duplicationError, fetchData: duplicateItem } = useAxios({
-    service: duplicateQuestion,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponseError: onDuplicateError,
-    onResponse: onDuplicateResponse
+  const { mutate: duplicateItem } = useMutation({
+    queryKey: ['questions', itemsPerPage, sort, page, selectedItems],
+    mutationFn: duplicateQuestion,
+    onSuccess: onDuplicateResponse,
+    onError: handleErrorAlert
   })
 
-  const handleDuplicate = async (itemId: string) => {
-    await duplicateItem(itemId)
-    if (!duplicationError) await fetchData()
-  }
+  useEffect(() => {
+    if (questionsError) {
+      handleErrorAlert(questionsError)
+    }
+  }, [handleErrorAlert, questionsError])
 
   const props = {
     columns: columnsToShow,
-    data: { response, getData: fetchData },
+    data: {
+      response: questions ?? defaultResponses.itemsWithCount,
+      getData: getQuestions
+    },
     services: {
       deleteService: deleteQuestion
     },
     itemsPerPage,
     actions: {
       onEdit: editQuestion,
-      onDuplicate: (itemId: string) => handleDuplicate(itemId)
+      onDuplicate: (itemId: string) => duplicateItem(itemId)
```

Comment:
Can you pass it directly?
```suggestion
      onDuplicate: duplicateItem
```